enumerable.rb

```diff
@@ -0,0 +1,125 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength,
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    n = 0
+    while n < to_a.length
+      yield(self[n]) if is_a?(Array)
+      yield([keys[n], values[n]]) if is_a?(Hash)
+      yield(to_a[n]) if is_a?(Range)
+      n += 1
+    end
+    self
+  end
+
+  def my_each_with_index(&block)
+    return to_enum(:my_each_with_index) unless block_given?
+
+    n = 0
+    while n < to_a.length
+      block.call(self[n], n) if is_a?(Array)
+      yield([keys[n], values[n]]) if is_a?(Hash)
+      block.call(to_a[n]) if is_a?(Range)
+      n += 1
```

Comment:
- [ ] **[REQUIRED]** The first argument for a hash shouldn't just be the keys. It's an array of the key as a symbol and its value.
    - Example:
        ```ruby
        enum = { a: 1, b: 2, c: 3, d: 4, e: 5 }
        my_each_with_index_output = ''
        block = proc { |num, idx| my_each_with_index_output += "Num: #{num}, idx: #{idx}\n" }
        enum.each_with_index(&block)
        puts my_each_with_index_output
        ```
        should return:
        ```ruby
        Num: [:a, 1], idx: 0
        Num: [:b, 2], idx: 1
        Num: [:c, 3], idx: 2
        Num: [:d, 4], idx: 3
        Num: [:e, 5], idx: 4
        ```
- [ ] **[REQUIRED]** In case of a `range` you haven't passed in `n` as the second parameter.